src/shogun/io/TBOutputFormat.cpp

```diff
@@ -46,21 +46,21 @@ using namespace shogun;
 
 #define CHECK_TYPE(type)                                                       \
 	else if (                                                                  \
-	    value.first.get_value().type_info().hash_code() ==                     \
```

Comment:
Are this macros still needed with the any dispatch?

Reply:
No, they are not needed. Will remove them.